desc/protoparse/linker_test.go

```diff
@@ -1017,6 +1017,17 @@ func TestLinkerValidation(t *testing.T) {
 			"foo.proto:4:3: field Foo.bar: custom JSON name \"foo\" conflicts with default JSON name of field foo, defined at foo.proto:3:3",
 		},
 		{
+			map[string]string{
+				"foo.proto": "syntax = \"proto3\";\n" +
+					"message Blah {\n" +
+					"  message Foo {\n" +
+					"    string foo = 1;\n" +
+					"    string bar = 2 [json_name=\"foo\"];\n" +
+					"  }\n" +
+					"}\n",
+			},
+			"foo.proto:5:5: field Foo.bar: custom JSON name \"foo\" conflicts with default JSON name of field foo, defined at foo.proto:4:5",
+		}, {
```

Comment:
This (and a couple others below) are new tests to verify that messages _nested in other messages_ are also correctly checked.